data-plane/receiver/src/test/java/dev/knative/eventing/kafka/broker/receiver/impl/IngressProducerReconcilableStoreTest.java

```diff
@@ -321,4 +323,130 @@ private void testStore(
     when(producer.unwrap()).thenReturn(new MockProducer<>());
     return producer;
   }
+
+  @Test
+  public void shouldResolvePathCorrectly(final Vertx vertx,
+                                         final VertxTestContext context) {
+    final var resource1 = DataPlaneContract.Resource.newBuilder()
+      .setUid("1")
+      .addTopics("topic1")
+      .setBootstrapServers("kafka-1:9092")
+      .setIngress(DataPlaneContract.Ingress.newBuilder().setPath("/hello1")
+        .setContentMode(DataPlaneContract.ContentMode.STRUCTURED).build())
+      .build();
+
+    final var resource2 = DataPlaneContract.Resource.newBuilder()
+      .setUid("2")
+      .addTopics("topic2")
+      .setBootstrapServers("kafka-2:9092")
+      .setIngress(DataPlaneContract.Ingress.newBuilder().setHost("http://host2").setPath("/")
+        .setContentMode(DataPlaneContract.ContentMode.STRUCTURED).build())
+      .build();
+
+    final var resource3 = DataPlaneContract.Resource.newBuilder()
+      .setUid("3")
+      .addTopics("topic3")
+      .setBootstrapServers("kafka-3:9092")
+      .setIngress(DataPlaneContract.Ingress.newBuilder().setHost("http://host3").setPath("/hello3")
+        .setContentMode(DataPlaneContract.ContentMode.STRUCTURED).build())
+      .build();
+
+    // additional resource to make sure path "/" don't mean anything
+    final var resource4 = DataPlaneContract.Resource.newBuilder()
+      .setUid("4")
+      .addTopics("topic4")
+      .setBootstrapServers("kafka-4:9092")
+      .setIngress(DataPlaneContract.Ingress.newBuilder().setHost("http://host4").setPath("/")
+        .setContentMode(DataPlaneContract.ContentMode.STRUCTURED).build())
+      .build();
+
+    KafkaProducer<String, CloudEvent> producer1 = mockProducer();
+    KafkaProducer<String, CloudEvent> producer2 = mockProducer();
+    KafkaProducer<String, CloudEvent> producer3 = mockProducer();
+    KafkaProducer<String, CloudEvent> producer4 = mockProducer();
+
+    final var store = new IngressProducerReconcilableStore(
+      AuthProvider.noAuth(),
+      new Properties(),
+      properties -> {
+        if ("kafka-1:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
+          return producer1;
+        } else if ("kafka-2:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
+          return producer2;
+        } else if ("kafka-3:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
+          return producer3;
+        } else if ("kafka-4:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
+          return producer4;
+        }
```

Comment:
else after return doesn't seem useful and we usually return early in this codebase if that's possible

```suggestion
        if ("kafka-1:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
          return producer1;
        }
        if ("kafka-2:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
          return producer2;
        }
        if ("kafka-3:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
          return producer3;
        }
        if ("kafka-4:9092".equals(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG))) {
          return producer4;
        }
```

Reply:
another maybe more maintainable possibility is to have a map bootrstrapServer -> producer and then just call:
```java
return m.get(properties.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG)));
```